lib/k8s-e2e-tests/tests/vector.rs

```diff
@@ -0,0 +1,109 @@
+use k8s_e2e_tests::*;
+use k8s_test_framework::{
+    lock, test_pod, vector::Config as VectorConfig, wait_for_resource::WaitFor,
+};
+
+const HELM_CHART_VECTOR: &str = "vector";
+
+const HELM_VALUES_STDOUT_SINK: &str = r#"
+vector-aggregator:
+  vectorSource:
+    sourceId: vector
+
+  sinks:
+    stdout:
+      type: "console"
+      inputs: ["vector"]
+      rawConfig: |
+        target = "stdout"
+        encoding = "json"
+"#;
+
+/// This test validates that vector picks up logs with an agent and
+/// delivers them to the aggregator out of the box.
+#[tokio::test]
+async fn logs() -> Result<(), Box<dyn std::error::Error>> {
+    let _guard = lock();
```

Comment:
This is a cool trick. I'm sure we're using it elsewhere in the k8s test code but I just wanted to mention it :)